tests/apollo/util/bft.py

```diff
@@ -1690,13 +1742,27 @@ async def wait_for_replica_to_reach_view(self, replica_id, expected_view):
         """
         Wait for a single replica to reach the expected view.
         """
-        with log.start_action(action_type="wait_for_replica_to_reach_view", replica=replica_id, expected_view=expected_view) as action:
+        with log.start_action(action_type="wait_for_replica_to_reach_view", replica=replica_id,
+                              expected_view=expected_view) as action:
             async def expected_view_to_be_reached():
-                key = ['replica', 'Gauges', 'view']        
+                key = ['replica', 'Gauges', 'view']
                 replica_view = await self.retrieve_metric(replica_id, *key)
-                
+
                 if replica_view is not None and replica_view == expected_view:
                     return replica_view
-        
+
         return await self.wait_for(expected_view_to_be_reached, 30, .5)
-                
+
+    @staticmethod
+    @lru_cache(maxsize=None)
+    def binary_digest(binary_path: str):
+        buf_size = 4096
+        sha1 = hashlib.sha1()
+
+        with open(binary_path, 'rb') as f:
+            while True:
+                data = f.read(buf_size)
+                if not data:
+                    break
+                sha1.update(data)
+        return sha1.hexdigest()
```

Comment:
For Apollo improvements, please open a different PR that explains what are the changes and why they are needed.